pkg/standalone-utils/contracts/test/MockSilo.sol

```diff
@@ -0,0 +1,111 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/ISilo.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+
+import "./MockShareToken.sol";
+
+contract MockBaseSilo is IBaseSilo {
+    address private immutable _siloAsset;
+    mapping(address => AssetStorage) internal _assetStorage;
+
+    constructor(address siloAsset) {
+        _siloAsset = siloAsset;
+    }
+
+    function assetStorage(address _asset) external view override returns (AssetStorage memory) {
+        return _assetStorage[_asset];
+    }
+
+    function setAssetStorage(
+        address interestBearingAsset,
+        IShareToken collateralToken,
+        IShareToken collateralOnlyToken,
+        IShareToken debtToken,
+        uint256 totalDeposits,
+        uint256 collateralOnlyDeposits,
+        uint256 totalBorrowAmount
+    ) external {
+        AssetStorage memory storageValue = AssetStorage(
+            collateralToken,
+            collateralOnlyToken,
+            debtToken,
+            totalDeposits,
+            collateralOnlyDeposits,
+            totalBorrowAmount
+        );
+
+        _assetStorage[interestBearingAsset] = storageValue;
+    }
+}
+
+contract MockSilo is ISilo, MockBaseSilo {
+    using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
+
+    uint256 public rate;
+
+    constructor(address _siloAsset) MockBaseSilo(_siloAsset) {
+        rate = FixedPoint.TWO;
+    }
+
+    function depositFor(
+        address _asset,
+        address _depositor,
+        uint256 _amount,
+        bool /*_collateralOnly*/
+    ) external override returns (uint256 collateralAmount, uint256 collateralShare) {
+        IERC20(_asset).safeTransferFrom(msg.sender, address(this), _amount);
+        address shareTokenAddress = address(_assetStorage[_asset].collateralToken);
+        uint256 shares = underlyingToShares(_amount);
+        MockShareToken(shareTokenAddress).mint(_depositor, shares);
+        return (_amount, shares);
+    }
+
+    function withdraw(
+        address _asset,
+        uint256 _amount,
+        bool /*_collateralOnly*/
+    ) external override returns (uint256 withdrawnAmount, uint256 withdrawnShare) {
+        address shareTokenAddress = address(_assetStorage[_asset].collateralToken);
+        uint256 burnedShare;
+        if (_amount == type(uint256).max) {
```

Comment:
Like the others, could be a constant, either MAX_UINT256 with a comment about what that means, or something self-documenting like ENTIRE_BALANCE_SENTINEL_VALUE

Reply:
Whats the benefit to having type(uint256).max being set as a constant? Wouldn't it cost more to make it a constant and read it during comparison operations vs leaving it as is and adding a comment explaining the logic.

Reply:
There's no cost to read a constant value (they're interpreted at compile-time and baked into the bytecode); it's just for documentation. If you only use it once I guess it doesn't matter if it's a constant vs a comment. A descriptive constant would be self-documenting (i.e., you wouldn't need to read the comment to understand it), but I don't feel strongly.